order/tradingfees.go

```diff
@@ -179,6 +179,10 @@ func takerDelta(feeSchedule terms.FeeSchedule,
 	balanceDelta := -satsPremium
 	takerFeesPaid := satsPremium
 
+	// If there is a self channel balance, it must also be paid out of the
+	// taker account.
+	balanceDelta -= selfChanBalance
```

Comment:
Good point, we definitely need to think about that. I think since they still pay the chain fee part (and execution base fee) it should be okay to do without an additional fee? Or we just need to bump the base fee itself.
Anyway, I think for a first iteration it's probably good to limit the `self_chan_balance` to the order amount itself.